### PR TITLE
Submit new diary card 1

### PR DESCRIPTION
**Pull request by NimeshPiyumantha**
[ITP-2266] [ITP-2269] [ITP-2275] [ITP-2278] [ITP-2290] [ITP-2284]

[ITP-2266]: https://fcodelabs.atlassian.net/browse/ITP-2266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2269]: https://fcodelabs.atlassian.net/browse/ITP-2269?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2275]: https://fcodelabs.atlassian.net/browse/ITP-2275?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2278]: https://fcodelabs.atlassian.net/browse/ITP-2278?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2290]: https://fcodelabs.atlassian.net/browse/ITP-2290?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2284]: https://fcodelabs.atlassian.net/browse/ITP-2284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Content/Content.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Route, Routes } from "react-router-dom";
+import SignInPage from "../../pages/SignInPage/SignInPage";
+import DiaryHome from "../../pages/DiaryHomePage/DiaryHome";
+
+export default function Content() {
+  return (
+    <Routes>
+      <Route path="/" element={<SignInPage />} />
```
Comment (shamal-iroshan) [inline src/components/Content/Content.tsx]: store routes in a constant file and use that, don't hardcode routes everywhere. and also rename this file as RootRouter or something.

File: src/components/DiaryCard/DiaryCard.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+import Grid from "@mui/material/Grid";
+
+interface DiaryCardProps {
```
Comment (shamal-iroshan) [inline src/components/DiaryCard/DiaryCard.tsx]: better to start the name of an interface with I. ex:- IDiaryCardProps

File: src/components/DiaryCard/DiaryCard.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  username,
+  description,
+}) => {
+  const [showMore, setShowMore] = useState(false);
+
+  const toggleShowMore = () => {
+    setShowMore(!showMore);
+  };
+
+  const cardStyle: React.CSSProperties = {
```
Comment (shamal-iroshan) [inline src/components/DiaryCard/DiaryCard.tsx]: don't use inline styles or separate CSS files when using MUI, use there styled utility to create custom components or use there sx prop for adding styles

File: src/pages/DiaryHomePage/DiaryHome.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+  const handleSubmit = () => {
+    if (!title.trim() || !description.trim()) {
+      setIsSnackbarOpen(true);
+      return;
+    }
+
+    const newEntry: DiaryEntry = {
+      title,
+      uName,
```
Comment (shamal-iroshan) [inline src/pages/DiaryHomePage/DiaryHome.tsx]: use meaningful names if this is a user name make it userName.

File: src/pages/DiaryHomePage/DiaryHome.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  const handleSnackbarClose = () => {
+    setIsSnackbarOpen(false);
+  };
+
+  const handleLogout = () => {
+    navigate("/");
+  };
+
+  return (
+    <ThemeProvider theme={theme}>
```
Comment (shamal-iroshan) [inline src/pages/DiaryHomePage/DiaryHome.tsx:69]: use theme provider and cssbaseline in root level not in component level

File: src/pages/SignInPage/SignInPage.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import { useNavigate } from "react-router-dom";
+import { ThemeProvider } from "@emotion/react";
+import { Box, CssBaseline, Grid } from "@mui/material";
+import { theme } from "../../theme/theme";
+
+const SignInPage: React.FC = () => {
+  const [name, setName] = useState<string>("");
+  const navigate = useNavigate();
+
+  const generateRandomName = () => {
```
Comment (shamal-iroshan) [inline src/pages/SignInPage/SignInPage.tsx]: move this function into util directory

File: src/pages/SignInPage/SignInPage.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    const randomNames = ["Akila", "Kavindu","Pasan", "Charlie", "David", "Eve"];
+    const randomIndex = Math.floor(Math.random() * randomNames.length);
+    setName(randomNames[randomIndex]);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    console.log(`Submitted name: ${name}`);
```
Comment (shamal-iroshan) [inline src/pages/SignInPage/SignInPage.tsx]: remove unwanted logs

File: src/pages/SignInPage/SignInPage.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    navigate("/home", { state: { name: name } });
+  };
+
+  const isSubmitDisabled = name.trim() === "";
+
+  return (
+    <ThemeProvider theme={theme}>
```
Comment (shamal-iroshan) [inline src/pages/SignInPage/SignInPage.tsx:23]: fix this themeprovider and cssbaseline as well

**Review by shamal-iroshan — CHANGES_REQUESTED** [review]
fix commented things

**Comment by NimeshPiyumantha** [comment]
Done

**Review by shamal-iroshan — APPROVED** [review]

**Pull request merged**